test/infrastructure/inmemory/internal/server/api/handler.go

```diff
@@ -274,7 +274,11 @@ func (h *apiServerHandler) apiV1Create(req *restful.Request, resp *restful.Respo
 	// TODO: consider check vs enforce for namespace on the object - namespace on the request path
 	obj.SetNamespace(req.PathParameter("namespace"))
 	if err := cloudClient.Create(ctx, obj); err != nil {
-		_ = resp.WriteErrorString(http.StatusInternalServerError, err.Error())
+		if status, ok := err.(apierrors.APIStatus); ok || errors.As(err, &status) {
+			_ = resp.WriteHeaderAndEntity(int(status.Status().Code), status)
```

Comment:
The answer to 1 and 2 is yes: after this fix now API errors generated inside the in-memory cache are encoded properly, and the client converts them back into API error structs so you can use API machinery function to check the error type. 
WRT to question 3, I re-used the error handling code that was already implemented for get, and it worked well, so I did not really went looking for alternatives or optimizations.

Reply:
Thx for the answers. In general I would prefer using the As function from the Go SDK instead of using one from an external library (AFAIK we usually always use the one from the Go SDK).

But I guess they both work, so no objection for the in memory provider

Reply:
I think the only reason why we use errors.As from pkg/errors instead of the Go SDK today is because we import pkg/errors as errors. So basically it's caused by mis-aliasing.